src/m365/aad/commands/group/group-list.ts

```diff
@@ -0,0 +1,49 @@
+import { Logger } from '../../../../cli';
+import GlobalOptions from '../../../../GlobalOptions';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import { Group } from '@microsoft/microsoft-graph-types';
+
+interface CommandArgs {
+  options: GlobalOptions;
+}
+
+class AadGroupListCommand extends GraphItemsListCommand<Group>   {
+  public get name(): string {
+    return commands.GROUP_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all groups defined in Azure Active Directory.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'groupTypes'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAllItems(`${this.resource}/v1.0/groups`, logger, true)
+      .then((): void => {
+        this.items.forEach(group => {
+          if (group.groupTypes && group.groupTypes.length > 0 && group.groupTypes[0] === "Unified") {
+            group.groupTypes = ["Microsoft 365"];
```

Comment:
Instead of rewriting the original data that we receive from the API, let's introduce a new property named `groupType` where we set the group display name and which we display in the text output. For JSON output we'd then have access to the original data as received from the API.